### ISSUE TITLE
pooling.py — validation, mask threading, and doctest coverage

### ISSUE THREAD

**Issue by stepanyanhayk** [McDermottHealthAI/MedRAP#56]
> **Depends on**: #51 (EncoderOutput.attention_mask) — the mask threading fix here cannot be completed until that field exists

## SWE fix

**`IdentityPooling.pool` doesn't validate `S == 1`**
`fused_state.squeeze(1)` on `(B, S, D)` with `S > 1` is a silent no-op — returns `(B, S, D)` instead of `(B, D)`, which then fails with a cryptic shape error inside the linear head rather than a clear message at the pooling stage. Add an explicit check:
```python
if fused_state.shape[1] != 1:
    raise ValueError(...)
```

## ML fix

**`model.py:308` never passes `attention_mask` to pooling**
```python
pooled = self.pooling(fusion_out.fused_state)  # mask silently dropped
```
`MaskedMeanPooling` falls back to unmasked mean when mask is `None`, so padding tokens are included in the pooled representation. Once `EncoderOutput.attention_mask` exists (#51), thread it through `model.py` to this call site.

## Tests

**Add all-masked edge case doctest to `MaskedMeanPooling.pool`**
`counts.clamp_min(1)` keeps the denominator safe when all positions are masked, returning a zero vector. Document this with a doctest consistent with how `PerDocMeanPooledRetrievalEncoder` covers the same edge case.

**Add error doctest to `IdentityPooling.pool`**
Once the `S == 1` validation is added, add a doctest showing the `ValueError` for `S > 1` input.
